### PR TITLE
Always run tests on main to make sure caches exist

### PR DESCRIPTION
**Pull request by ForsakenHarmony**

### CODE REVIEW

**Comment by ForsakenHarmony** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3245** <a href="https://app.graphite.dev/github/pr/vercel/turbo/3245" target="_blank">Graphite</a> 
    * **PR #3246** <a href="https://app.graphite.dev/github/pr/vercel/turbo/3246" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/vercel/turbo/3246?utm_source=stack-comment).

**Review by wbinnssmith — APPROVED** [review]

**Review by sokra — CHANGES_REQUESTED** [review]
That doesn't seem like a good solution. Main should also run only things that are affected by the commit.

File: .github/workflows/pr-clean-caches.yml [hunk 1]
```diff
@@ -16,9 +17,6 @@ jobs:
     runs-on: ubuntu-latest
     if: ${{ github.ref != 'refs/heads/main' }}
```
Comment (sokra) [inline .github/workflows/pr-clean-caches.yml:18]: Why does this workflow run on PR when it has condition like that

Comment (ForsakenHarmony) [reply .github/workflows/pr-clean-caches.yml:18]: What do you mean? it's a `!=`

**Comment by ForsakenHarmony** [comment]
@sokra do you have a better solution?

I don't see the downside except for spending a bit more CI time on main, but in return we should get less CI time in PRs

File: .github/workflows/pr-clean-caches.yml (hunk 1 above)
Comment (sokra) [reply .github/workflows/pr-clean-caches.yml:18]: Ah I see

**Comment by sokra** [comment]
> do you have a better solution?

The cache should be still up-to-date when nothing has changed. I see no reason why caching won't work correctly without that. Maybe there is some other bug that break the cache.

**Comment by tknickman** [comment]
@ForsakenHarmony is this still needed?

**Pull request closed without merging**